src/main/java/ch/njol/skript/effects/EffOpenInventory.java

```diff
@@ -34,125 +36,201 @@
+			this.methodExists = methodExists;
+			this.methodError = methodError;
+			this.property = property;
+		}
+
+		private String getFormatted() {
+			return this.toString().toLowerCase(Locale.ENGLISH) + ":" + property;
+		}
+
+		private Version getVersion() {
+			return version;
+		}
+
+		private boolean doesMethodExist() {
+			return methodExists;
+		}
+
+		private String getMethodError() {
+			return methodError;
+		}
+
+		private static String construct() {
+			StringBuilder builder = new StringBuilder("(");
+			OpenableInventorySyntax[] values = OpenableInventorySyntax.values();
+			for (int i = 0; i < values.length; i++ ) {
+				builder.append(values[i].getFormatted());
+				if (i + 1 < values.length)
+					builder.append("|");
+			}
+			return builder.append(")").toString();
+		}
+	}
+
 	static {
 		Skript.registerEffect(EffOpenInventory.class,
-				"(open|show) ((0¦(crafting [table]|workbench)|1¦chest|2¦anvil|3¦hopper|4¦dropper|5¦dispenser) (view|window|inventory|)|%-inventory/inventorytype%) (to|for) %players%",
-				"close [the] inventory [view] (to|of|for) %players%", "close %players%'[s] inventory [view]");
+				"show %inventory/inventorytype% (to|for) %players%",
+				"open [a[n]] " + OpenableInventorySyntax.construct() + "[view|window|inventory] (to|for) %players%",
+				"close [the] inventory [view] (of|for) %players%",
+				"close %players%'[s] inventory [view]");
 	}
-	
+
+	private boolean open;
+
 	@Nullable
-	private Expression<?> invi;
-	
-	boolean open;
-	private int invType;
-	
-	@SuppressWarnings("null")
+	private OpenableInventorySyntax syntax;
+
+	@Nullable
+	private Expression<?> inventory;
 	private Expression<Player> players;
-	
-	@SuppressWarnings({"unchecked", "null"})
+
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		int openFlag = 0;
-		if(parseResult.mark >= 5) {
-			openFlag = parseResult.mark ^ 5;
-			invType = DISPENSER;
-		} else if(parseResult.mark >= 4) {
-			openFlag = parseResult.mark ^ 4;
-			invType = DROPPER;
-		} else if(parseResult.mark >= 3) {
-			openFlag = parseResult.mark ^ 3;
-			invType = HOPPER;
-		} else if (parseResult.mark >= 2) {
-			openFlag = parseResult.mark ^ 2;
-			invType = ANVIL;
-		} else if (parseResult.mark >= 1) {
-			openFlag = parseResult.mark ^ 1;
-			invType = CHEST;
-		} else if (parseResult.mark >= 0) {
-			invType = WORKBENCH;
-			openFlag = parseResult.mark ^ 0;
-		} else {
-			openFlag = parseResult.mark;
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (matchedPattern == 1) {
+			open = true;
+			assert !parseResult.tags.isEmpty() : "Syntax incorrectly handled parse result tags.";
+			syntax = OpenableInventorySyntax.valueOf(parseResult.tags.get(0).toUpperCase(Locale.ENGLISH));
+			if (!Skript.isRunningMinecraft(syntax.getVersion())) {
```

Comment:
It probably won't cause an error, but getVersion is nullable and thus a null check should be performed (same goes for doesMethodExist)